lib/daru/io/base.rb

```diff
@@ -0,0 +1,16 @@
+module Daru
+  module IO
+    class Base
+      def initialize(binding)
+        args = method(__method__).parameters.map do |_, name|
+          [name, binding.local_variable_get(name.to_s)]
+        end.to_h
+
+        args.each do |k, v|
+          instance_variable_set("@#{k}", v)
+          define_singleton_method(k) { instance_variable_get("@#{k}") }
+        end
+      end
```

Comment:
Hmm, let's discuss about this on issue #19. Depending on the consensus achieved, I'll revert this part if required. 🙂